### PR TITLE
SW-4662 More fixedMenu fixes, handle scroll

### PR DESCRIPTION
**Pull request by karthikbtf**
- previously the fixedMenu will be fixed during scroll and separated from input component
- added a scroll event handler to reposition menu
- updated story to enable scrolling

#### Before (without scroll handling)
Select - Editable ⋅ Storybook 2024-04-09 11-34-45

#### After (after scroll handling)
Select - Editable ⋅ Storybook 2024-04-09 11-34-01

### CODE REVIEW

File: src/components/Select/SelectT.tsx [hunk 1]
```diff
@@ -81,6 +81,33 @@ export default function SelectT<T>(props: SelectTProps<T>): JSX.Element {
   const inputRef = useRef<HTMLDivElement>(null);
   const dropdownRef = useRef<HTMLUListElement>(null);
 
+  const hasOptions = useMemo<boolean>(() => options !== undefined &&  options.length > 0, [options]);
+
+  const repositionMenu = useCallback(() => {
```
Comment (karthikbtf) [inline src/components/Select/SelectT.tsx:86]: These were moved up in order and converted to a useCallback. No other code changes here.

File: src/components/Select/SelectT.tsx [hunk 2]
```diff
@@ -120,34 +141,24 @@ export default function SelectT<T>(props: SelectTProps<T>): JSX.Element {
...
-            dropdownRef.current.clientHeight - (dropdownBottom - windowHeightThreshold)
-          }px`;
-        }
-        setFixedMenuPositioned(true);
-      }
-    } else if (fixedMenu) {
-      setFixedMenuPositioned(false);
-    }
-  }, [fixedMenu, openedOptions, hasOptions]);
+    window.addEventListener('click', handleClick);
```
Comment (karthikbtf) [inline src/components/Select/SelectT.tsx:151]: This was moved down to enable declaration of useCallback dependencies, absence of which was causing behavior issues.

**Comment by karthikbtf** [comment]
publish rc

**Comment by karthikbtf** [comment]
@constanzauanini can you check this out when you have time? More context with fixedMenu.

**Comment by constanzauanini** [comment]
@karthikbtf I tested on vercel story, and when opening a menu of an input that is down and then scrolling up (with the menu opened), and then scrolling back down again to the input, the menu is smaller and doesn't show all options. But if you close it and open again, it gets fixed. Anyways, the menu is right positioned now, that other usecase it's too specific

**Review by constanzauanini — APPROVED** [review]
One behavioural comment, but not blocking for me

**Comment by karthikbtf** [comment]
> @karthikbtf I tested on vercel story, and when opening a menu of an input that is down and then scrolling up (with the menu opened), and then scrolling back down again to the input, the menu is smaller and doesn't show all options. But if you close it and open again, it gets fixed. Anyways, the menu is right positioned now, that other usecase it's too specific

Thanks for taking a look @constanzauanini , do you have time to take a video of what you see? I can't seem to repro it. I'd prefer to fix it if I can.

**Comment by constanzauanini** [comment]
> > @karthikbtf I tested on vercel story, and when opening a menu of an input that is down and then scrolling up (with the menu opened), and then scrolling back down again to the input, the menu is smaller and doesn't show all options. But if you close it and open again, it gets fixed. Anyways, the menu is right positioned now, that other usecase it's too specific
> 
> Thanks for taking a look @constanzauanini , do you have time to take a video of what you see? I can't seem to repro it. I'd prefer to fix it if I can.

Here is the video, the options are all there, but the menu's height is smaller:

Select - Docs ⋅ Storybook

**Comment by karthikbtf** [comment]
> > > @karthikbtf I tested on vercel story, and when opening a menu of an input that is down and then scrolling up (with the menu opened), and then scrolling back down again to the input, the menu is smaller and doesn't show all options. But if you close it and open again, it gets fixed. Anyways, the menu is right positioned now, that other usecase it's too specific
> > 
> > 
> > Thanks for taking a look @constanzauanini , do you have time to take a video of what you see? I can't seem to repro it. I'd prefer to fix it if I can.
> 
> Here is the video, the options are all there, but the menu's height is smaller:
> 
> Select - Docs ⋅ Storybook [ Select - Docs ⋅ Storybook ](https://private-user-images.githubusercontent.com/5919083/321017857-31870e8c-c769-4001-827d-7e7c49ad0d2b.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI3MDIzODYsIm5iZiI6MTcxMjcwMjA4NiwicGF0aCI6Ii81OTE5MDgzLzMyMTAxNzg1Ny0zMTg3MGU4Yy1jNzY5LTQwMDEtODI3ZC03ZTdjNDlhZDBkMmIuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDQwOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA0MDlUMjIzNDQ2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZjkzMTBlZmY1NzFjZDFjZGJiMGRiYjZlODRhOWU0ODY2NGRhYTEyZjI5MjJiNTNkZGU4MzZjMWRkY2YwNmZmNyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.RCzCt8f1n92eGJQ0f0R_nvt6UuwNB8S5akvyOE3Gx38) [ ](https://private-user-images.githubusercontent.com/5919083/321017857-31870e8c-c769-4001-827d-7e7c49ad0d2b.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI3MDIzODYsIm5iZiI6MTcxMjcwMjA4NiwicGF0aCI6Ii81OTE5MDgzLzMyMTAxNzg1Ny0zMTg3MGU4Yy1jNzY5LTQwMDEtODI3ZC03ZTdjNDlhZDBkMmIuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDQwOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA0MDlUMjIzNDQ2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZjkzMTBlZmY1NzFjZDFjZGJiMGRiYjZlODRhOWU0ODY2NGRhYTEyZjI5MjJiNTNkZGU4MzZjMWRkY2YwNmZmNyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.RCzCt8f1n92eGJQ0f0R_nvt6UuwNB8S5akvyOE3Gx38)

Got it thanks. You are looking at the docs version. Can you click on the `Editable` story on the left menu and try again? Not sure how the docs version works but the actual standalone story seems to be ok.

**Comment by constanzauanini** [comment]
> > > > @karthikbtf I tested on vercel story, and when opening a menu of an input that is down and then scrolling up (with the menu opened), and then scrolling back down again to the input, the menu is smaller and doesn't show all options. But if you close it and open again, it gets fixed. Anyways, the menu is right positioned now, that other usecase it's too specific
> > > 
> > > 
> > > Thanks for taking a look @constanzauanini , do you have time to take a video of what you see? I can't seem to repro it. I'd prefer to fix it if I can.
> > 
> > 
> > Here is the video, the options are all there, but the menu's height is smaller:
> 
> Got it thanks. You are looking at the docs version. Can you click on the `Editable` story on the left menu and try again? Not sure how the docs version works but the actual standalone story seems to be ok.

I can't scroll in the Editable story when the menu is opened

**Comment by karthikbtf** [comment]
> > > > > @karthikbtf I tested on vercel story, and when opening a menu of an input that is down and then scrolling up (with the menu opened), and then scrolling back down again to the input, the menu is smaller and doesn't show all options. But if you close it and open again, it gets fixed. Anyways, the menu is right positioned now, that other usecase it's too specific
> > > > 
> > > > 
> > > > Thanks for taking a look @constanzauanini , do you have time to take a video of what you see? I can't seem to repro it. I'd prefer to fix it if I can.
> > > 
> > > 
> > > Here is the video, the options are all there, but the menu's height is smaller:
> > 
> > 
> > Got it thanks. You are looking at the docs version. Can you click on the `Editable` story on the left menu and try again? Not sure how the docs version works but the actual standalone story seems to be ok.
> 
> I can't scroll in the Editable story when the menu is opened

I had to open the js console to enable some scrolling.

**Comment by karthikbtf** [comment]
> > > > > > @karthikbtf I tested on vercel story, and when opening a menu of an input that is down and then scrolling up (with the menu opened), and then scrolling back down again to the input, the menu is smaller and doesn't show all options. But if you close it and open again, it gets fixed. Anyways, the menu is right positioned now, that other usecase it's too specific
> > > > > 
> > > > > 
> > > > > Thanks for taking a look @constanzauanini , do you have time to take a video of what you see? I can't seem to repro it. I'd prefer to fix it if I can.
> > > > 
> > > > 
> > > > Here is the video, the options are all there, but the menu's height is smaller:
> > > 
> > > 
> > > Got it thanks. You are looking at the docs version. Can you click on the `Editable` story on the left menu and try again? Not sure how the docs version works but the actual standalone story seems to be ok.
> > 
> > 
> > I can't scroll in the Editable story when the menu is opened
> 
> I had to open the js console to enable some scrolling.

Don't worry about this. Thanks for testing it.

**Comment by constanzauanini** [comment]
> Visit Preview

Inside the box I can't scroll if the menu is opened. But when adding space outside the box, I'm still seeing same issue:

Select - Editable ⋅ Storybook

**Comment by karthikbtf** [comment]
> > Visit Preview
> 
> Inside the box I can't scroll if the menu is opened. But when adding space outside the box, I'm still seeing same issue:
> 
> Select - Editable ⋅ Storybook [ Select - Editable ⋅ Storybook ](https://private-user-images.githubusercontent.com/5919083/321023611-a6ce3920-681c-44c9-b70e-761069ba6454.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI3MDQ1MTMsIm5iZiI6MTcxMjcwNDIxMywicGF0aCI6Ii81OTE5MDgzLzMyMTAyMzYxMS1hNmNlMzkyMC02ODFjLTQ0YzktYjcwZS03NjEwNjliYTY0NTQuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDQwOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA0MDlUMjMxMDEzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZmE2OWRjYWQyODliOWNiYjk0MzRjMGJmZWE0MmU1MWM2MjFiZDA3MWQ1ZDYwMjRiMWRmYWQ5YjVjZDE4YTdkOSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.HJPXBzXZfmqz0BmsvJtNG-U7axGWFKsxxh6uoZ_2XQg) [ ](https://private-user-images.githubusercontent.com/5919083/321023611-a6ce3920-681c-44c9-b70e-761069ba6454.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI3MDQ1MTMsIm5iZiI6MTcxMjcwNDIxMywicGF0aCI6Ii81OTE5MDgzLzMyMTAyMzYxMS1hNmNlMzkyMC02ODFjLTQ0YzktYjcwZS03NjEwNjliYTY0NTQuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDQwOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA0MDlUMjMxMDEzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZmE2OWRjYWQyODliOWNiYjk0MzRjMGJmZWE0MmU1MWM2MjFiZDA3MWQ1ZDYwMjRiMWRmYWQ5YjVjZDE4YTdkOSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.HJPXBzXZfmqz0BmsvJtNG-U7axGWFKsxxh6uoZ_2XQg)

Ugh ok, let me try again separately.

**Comment by karthikbtf** [comment]
> > Visit Preview
> 
> Inside the box I can't scroll if the menu is opened. But when adding space outside the box, I'm still seeing same issue:
> 
> Select - Editable ⋅ Storybook [ Select - Editable ⋅ Storybook ](https://private-user-images.githubusercontent.com/5919083/321023611-a6ce3920-681c-44c9-b70e-761069ba6454.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI3MDU1OTEsIm5iZiI6MTcxMjcwNTI5MSwicGF0aCI6Ii81OTE5MDgzLzMyMTAyMzYxMS1hNmNlMzkyMC02ODFjLTQ0YzktYjcwZS03NjEwNjliYTY0NTQuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDQwOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA0MDlUMjMyODExWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ODdjMDAxNzQxNTNmZjU1NDE4NmQ4MTI3YzJkN2JiODEzYTViMDg1YjE0YjMzZGY2NjNjMGUxMjgyNWQ1OTQ5ZCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.h3V2-PlXJN1e8pyagLRItwx_H8FDPv8fd8qqMBMmo8Y) [ ](https://private-user-images.githubusercontent.com/5919083/321023611-a6ce3920-681c-44c9-b70e-761069ba6454.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI3MDU1OTEsIm5iZiI6MTcxMjcwNTI5MSwicGF0aCI6Ii81OTE5MDgzLzMyMTAyMzYxMS1hNmNlMzkyMC02ODFjLTQ0YzktYjcwZS03NjEwNjliYTY0NTQuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDQwOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA0MDlUMjMyODExWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ODdjMDAxNzQxNTNmZjU1NDE4NmQ4MTI3YzJkN2JiODEzYTViMDg1YjE0YjMzZGY2NjNjMGUxMjgyNWQ1OTQ5ZCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.h3V2-PlXJN1e8pyagLRItwx_H8FDPv8fd8qqMBMmo8Y)

Ok looks like this is existing [code](https://github.com/terraware/web-components/blob/main/src/components/Select/SelectT.tsx#L95) which resizes the dropdown if there isn't enough height.
I added a check to not resize if this is part of a scroll (instead of a window resize).
New PR is out https://github.com/terraware/web-components/pull/507
